tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -823,10 +823,33 @@ def complex_agg(x, agg):
     res = [complex_agg(fft[config["coeff"]], config["attr"]) if config["coeff"] < len(fft)
            else np.NaN for config in param]
     index = ['coeff_{}__attr_"{}"'.format(config["coeff"], config["attr"]) for config in param]
+
+    # Spectral Centroid:
+    fft_abs = np.abs(fft)
+    res += [fft_abs[:len(x)//2].dot(np.arange(0, len(x)//2, 1)) / fft_abs[:len(x)//2].sum()]
+    index += ['centroid']
+
     return zip(index, res)
 
 
 @set_property("fctype", "simple")
+def spectral_centroid(x):
```

Comment:
we could even have a more generalized feature calculator "aggregated_fft", which performs aggregations on the fft coefficient.

One of those is the spectral centroid, others could be variance?

Reply:
maybe you got some more ideas @aloosley ?

Reply:
Yes.  I would support this.  I am currently trying to reproduce some old results that a few (apparently unknown) students made before me.  Allegedly they measured several FT moments including the variance, skew, and kurtosis from the fft and used them to discriminate between vibration types.  

Unfortunately, nothing is documented nor do I have access to their implementation.  Given my goals, I probably won't implement these in the near-term but if anybody else gets to it first, note the following.  Apparently there are useful and less-useful means of calculating them.  For example, spectral kurtosis could be implemented based on this document: https://hal.archives-ouvertes.fr/hal-00021302/document

Reply:
> Unfortunately, nothing is documented nor do I have access to their implementation. Given my goals, I probably won't implement these in the near-term but if anybody else gets to it first, note the following. Apparently there are good and less-good ways of calculating moments from the fft.

you won't have to implement them. just take the numpy implementation of variance, skew, kurtosis etc. and apply them to the fft vector. most work will be to write the unit tests I guess.

Reply:
Yes, you are correct that one can leverage scipy.stats to grab the higher order moments of any distribution.  

However, suffice it to say there are apparently a number of subtleties in calculating these features with respect to their physical interpretation.  First, depending on the need, it doesn't seem necessarily best that one should ignore the complex angular part of the fourier spectrum (Ref [1]).  Second, apparently there is some variance in how people implement spectral kurtosis.  Ref [2]'s implementation is slightly different (by an additive factor be a big deal) from scipy's two implementations.  I would have to give some thought to these implementations before feeling comfortable pull requesting them into tsfresh.

Refs:
[1] https://hal.archives-ouvertes.fr/hal-00021302/document
[2] https://de.mathworks.com/matlabcentral/fileexchange/48914-welch-s-estimate-of-the-spectral-kurtosis

But yes, I naïvely think somebody for the time being could just use scipy.stats's implementation and go with it.

Reply:
>  I would have to give some thought to these implementations before feeling comfortable pull requesting them into 

👍